src/fmeval/eval_algorithms/faithfulness.py

```diff
@@ -0,0 +1,297 @@
+        # TODO: handle edge case that num_statements is 0?
+
+        # calculate score
+        final_answer = "Final verdicts in order:"
+        final_answer = final_answer.lower()
+        if output.find(final_answer) != -1:
+            output = output[output.find(final_answer) + len(final_answer) :]
+            score = float(sum(1 if "yes" in answer else 0 for answer in output.strip().split(".") if answer != ""))
+            score = score / num_statements
+        else:
+            score = float(max(0, output.count("verdict: yes")) / num_statements)
+
+        return score
+
+
+class GetStatements(Transform):
+    """This transform invokes the judge model string from raw statements."""
+
+    def __init__(self, input_key: str, output_key: str, judge_model: ModelRunner):
+        """GetStatements Initializer.
+        :param input_key: The key corresponding to prompt to get statements.
+        :param output_key: The key corresponding to the statements, which gets added to the record.
+        :param judge_model: An instance of ModelRunner representing the judge model to be used.
+            If this argument is None, default judge model will be provided.
+        """
+        super().__init__(input_key, output_key, judge_model)
+        self.register_input_output_keys(
+            input_keys=[input_key],
+            output_keys=[output_key],
+        )
+        self.input_key = input_key
+        self.output_key = output_key
+        self.judge_model = judge_model
+
+    @validate_call
+    def __call__(self, record: Dict[str, Any]) -> Dict[str, Any]:
+        """Invokes the judge model to generate statements and arguments the input record with the statements.
+        :param record: The input record.
+        :returns: The input record with the mean added in.
+        """
+        get_raw_statements = GetModelOutputs(
+            input_to_output_keys={self.input_key: ["raw_statements"]},
+            model_runner=self.judge_model,
+        )
+        record_with_raw_statements = get_raw_statements(record)
+        raw_statements = record_with_raw_statements["raw_statements"].split("\n")
+        list_statements = [statement for statement in raw_statements if statement.startswith("Statement:")]
+        statements_str: str = "\n".join([f"{i + 1}.{st}" for i, st in enumerate(list_statements)])
+        record[self.output_key] = statements_str
+        return record
+
+
+class Faithfulness(EvalAlgorithmInterface):
+    """ """
+
+    eval_name = EvalAlgorithm.FAITHFULNESS.value
+
+    def __init__(self):
+        """Faithfulness initializer."""
+        super().__init__(EvalAlgorithmConfig())
+
+    @staticmethod
+    def _build_pipeline(
+        judge_model: ModelRunner,
+        long_form_prompt_template: str,
+        nli_statements_prompt_template: str,
+    ) -> TransformPipeline:
+        gen_long_form_prompt = GeneratePrompt(
+            input_keys=[],
+            output_keys=["long_form_prompt"],
```

Comment:
Can we please use constants (local to this module) for the various keys that are used by this eval algo? E.g. "long_form_prompt", "question", "answer", etc